README.md

```diff
@@ -20,16 +20,56 @@ async fn dyn_dispatch(iter: &mut DynNext<'_, i32>) {
     }
 }
 
-let a = [1, 2, 3];
-dyn_dispatch(DynNext::from_mut(&mut a.into_iter())).await;
+let v = [1, 2, 3];
+dyn_dispatch(&mut DynNext::boxed(my_next_iter).await;
+dyn_dispatch(DynNext::from_mut(&mut my_next_iter)).await;
 ```
 
-The general rule is that anywhere you would write `dyn Trait` (which would
+Where `my_next_iter` is a value of a type that you would create that implements `Next`. For example:
+
+```rust,ignore
+struct AsyncIter<T: Copy, const N: usize> {
```

Comment:
The const on this is just distracting, consider using a vec instead. Also, the struct has a different name from the impl block below.

If we're going to flesh this example out I say let's go ahead and make it a complete one that compiles.